core/config/env.go

```diff
@@ -0,0 +1,17 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// HomeDirFromEnvironment returns the directory to use
+// for reading config and storing variable data.
+// It returns $CHAIN_CORE_HOME,
+// or, if that is empty, $HOME/.chaincore.
+func HomeDirFromEnvironment() string {
```

Comment:
It looks like there's no corresponding delete of this function from chain/core?

Also, maybe consider moving this function separately in its own PR first?